mne/viz/_brain/_brain.py

```diff
@@ -2644,6 +2644,14 @@ def add_volume_labels(
         vox_mri_t = aseg.header.get_vox2ras_tkr()
         mult = 1e-3 if self._units == "m" else 1
         vox_mri_t[:3] *= mult
+        ornt = nib.orientations.axcodes2ornt(
+            nib.orientations.aff2axcodes(aseg.affine)
+        ).astype(int)
+        ras_ornt = nib.orientations.axcodes2ornt("RAS")
+        ornt_trans = nib.orientations.ornt_transform(ornt, ras_ornt)
+        aseg_data = nib.orientations.apply_orientation(aseg_data, ornt_trans)
+        aff_trans = nib.orientations.inv_ornt_aff(ornt_trans, aseg.shape)
+        vox_mri_t = np.dot(vox_mri_t, aff_trans)
```

Comment:
I am confused why this is necessary -- the `vox_mri_t` should already take care of putting everything in MRI surface RAS coordinates for us. Also, there are a number of places we use `vox_mri_t` and don't use any `axcodes2ornt` stuff, so it's weird this is the one place we need to do it.

Reply:
... to me it suggests maybe there is some bug with how `_marching_cubes` works or something, rather than some adjustment we need to make to `vox_mri_t`.

I would investigate deeper by using `freeview` and some specific voxel `i, j, k` in some known subvolume and what their viewer says for MRI surface RAS `x, y, z`, confirm that this is just `(i, j, k) @ vox_mri_t` (it had better be!), then inspect the outputs at various points of this function for that label. It should tell us where the problem is...

Reply:
So it looks like it was caused by the fortran order check in `_marching_cubes`. I think the solution I just pushed works, here is one issue where the order needed to be fortran https://github.com/mne-tools/mne-python/issues/10860 which lead to the order being auto-detected with `"A"`. Checking to be sure that that works now too.